asnake/utils/__init__.py

```diff
@@ -213,6 +206,20 @@ def is_expired(date):
     return False
 
 
+def is_expired(date):
+    """Takes a date and then checks whether today's date is before or after passed date.
+
+    :param string date: a representation of a date
+
+    :returns: False if date argument is after today. True otherwise
+    :rtype: bool
+    """
+    today = datetime.now()
+    date = date if date else datetime.strftime("%Y-%m-%d")
```

Comment:
Hmmm. I somehow passed over this when I reviewed before, but I think this function needs to be rewritten.

Specifically, I'm not sure what the guard around date is _for_; you can't omit the date argument, explicitly passing non-string falsey values violates the expectations declared in the doc, and passing the empty string into the function would, if I'm reading the intent right, set the date to today, making the function return True.  This seems like unexpected behavior.

What I'd expect the code in this function to look like is:
```python
from datetime import date # change include line bc date is what we actually care about 

def is_expired(date):
    # this throws value error, if handed a broken date
    # should be either documented or caught and handled
    parsed_date = date.fromisoformat(date)
    return parsed_date <= date.today()
```

Reply:
@p-galligan This was meant to be in the review below, it's the more significant thing by FAR than the one that actually went into the review